### PR TITLE
feat: ExaTrkX edge building KDTree on CPU + fixes + refactor + tests

### PR DESCRIPTION
**Pull request by benjaminhuth**
* Replaces brute force edge building with `Acts::KDTree` method
* Abstracts `std::vector` to `torch::Tensor` conversions
* Fixes bug in edge duplicate removal
* Add more unit tests and enable in CI

### CODE REVIEW

File: Plugins/ExaTrkX/include/Acts/Plugins/ExaTrkX/detail/buildEdges.hpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+namespace at {
+class Tensor;
+}
+
+namespace Acts {
+namespace detail {
+
+/// Post process edges
+at::Tensor postprocessEdgeTensor(at::Tensor edges, bool removeSelfLoops = true,
```
Comment (andiwand) [inline Plugins/ExaTrkX/include/Acts/Plugins/ExaTrkX/detail/buildEdges.hpp:21]: `edges` by value on purpose? or is this more like a pointer type?

File: Plugins/ExaTrkX/src/buildEdges.cpp [hunk 2]
```diff
@@ -119,192 +162,122 @@ torch::Tensor buildEdgesFRNN(at::Tensor &embedFeatures, int64_t numSpacepoints,
...
-    return std::sqrt(((a - b) * (a - b)).sum().item().to<float>());
-  };
+  auto operator[](std::size_t i) const { return ptr[i]; }
+};
 
-  struct TwoRanges {
-    int i_min, i_max, j_min, j_max;
-  };
+template <std::size_t Dim>
+auto dist(const Span<float, Dim> &a, const Span<float, Dim> &b) {
```
Comment (andiwand) [inline Plugins/ExaTrkX/src/buildEdges.cpp:204]:
```suggestion
float dist(const Span<float, Dim> &a, const Span<float, Dim> &b) {
```

**Review by andiwand — DISMISSED** [review]
took a quick look - looks good - nothing critical from my side

File: Plugins/ExaTrkX/include/Acts/Plugins/ExaTrkX/detail/buildEdges.hpp (hunk 1 above)
Comment (benjaminhuth) [reply Plugins/ExaTrkX/include/Acts/Plugins/ExaTrkX/detail/buildEdges.hpp:21]: `at::Tensor` is reference counted like `std::shared_ptr`, so I would keep it like this for now.

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results
### :red_circle: Some tests have failed!
Please investigate the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/6059086)!

|status|job|
|------|---|
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31790896) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsWorkflow.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31790895) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateTracks.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31790894) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsCoreSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31790893) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateActsSpacePoints.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31790892) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31790891) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateOrthogonalSeeds.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31790890) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsValidateClusters.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31790889) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsPersistifyEDM.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31790888) |
|:green_circle: | [run_ci_tests: ../athena/AtlasTest/CITest/test/ActsKfRefitting.sh](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31790887) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsExtrapolationAlgTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31790886) |
|:green_circle: | [run_ci_tests: python3 ../athena/Tracking/Acts/ActsGeometry/test/ActsITkTest.py](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31790885) |
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31790884) |
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31790883) |
|:red_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31790882) |
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31790881) |
|:red_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/31790880) |

**Pull request merged**